battle-service/src/main/java/com/company/battle/repositories/RoomRepository.java

```diff
@@ -4,6 +4,11 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
+import java.util.Optional;
+import java.util.UUID;
+
 @Repository
-public interface RoomRepository extends JpaRepository<RoomEntity, Integer> {
+public interface RoomRepository extends JpaRepository<RoomEntity, UUID> {
```

Comment:
What is room?

Reply:
Room is something like a waiting room while the search for an opponent is in progress. When the second player is found, the game begins.

Reply:
I believe it can be threat by game status